browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -401,6 +413,46 @@ void RewardsInternalsDOMHandler::OnGetEventLogs(ledger::type::EventLogs logs) {
       std::move(data));
 }
 
+void RewardsInternalsDOMHandler::GetAdDiagnostics(const base::ListValue* args) {
+  if (!ads_service_) {
+    NOTREACHED();
+    return;
+  }
+
+  ads_service_->GetAdDiagnostics(
+      base::BindOnce(&RewardsInternalsDOMHandler::OnGetAdDiagnostics,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void RewardsInternalsDOMHandler::OnGetAdDiagnostics(const bool success,
+                                                    const std::string& json) {
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::Value diagnostics(base::Value::Type::LIST);
+  if (success && !json.empty()) {
+    absl::optional<base::Value> serialized_json = base::JSONReader::Read(json);
+    if (serialized_json && serialized_json->is_list() &&
+        !serialized_json->GetList().empty()) {
+      diagnostics = std::move(*serialized_json);
+    }
+  }
+
+#if DCHECK_IS_ON()
+  DCHECK(diagnostics.is_list()) << "Diagnostics should be a list";
+  for (const auto& entry : diagnostics.GetList()) {
+    DCHECK(entry.is_dict()) << "Diagnostics entry should be a dictionary";
+    DCHECK(entry.FindKey("key")) << "Diagnostics entry should has 'key' key";
```

Comment:
nit: `Diagnostics entry should have 'key' key`

Reply:
done